pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -492,6 +492,79 @@ export default class SettingsValidator {
     return false;
   }
 
+  protected checkExtensions(
+    mergedSettings: Settings,
+    currentValue: Record<string, string>,
+    desiredValue: any,
+    errors: string[],
+    fqname: string,
+  ): boolean {
+    if (typeof desiredValue !== 'object' || !desiredValue) {
+      errors.push(`${ fqname }: "${ desiredValue }" is not a valid mapping`);
+
+      return false;
+    }
+
+    // makeRE is a tagged template for making regular expressions with /x (i.e.
+    // ignoring any whitespace within the regular expression itself).
+    function makeRE(strings: {raw: readonly string[]}, ...subsitutions: any[]) {
```

Comment:
typo: should be `substitutions`

Reply:
Also I have to ask, so it's probably worth commenting on what exactly `strings` is, because I'm not parsing the `{raw: readonly string[]}` type annotation

Reply:
Just a note that `raw` is not the best name to allude to everything this method does

Reply:
Unfortunately, it's called `raw` because that's what the JS API passes in. (In the mean time, this has been updated to use the `TemplateStringsArray` interface directly, to better show that we don't define the shape of this argument.)